src/internal/pjsdb/pjsdb.go

```diff
@@ -0,0 +1,120 @@
+// Package pjsdb exports functions and structs for creating, reading, writing,
+// and deleting Job and Queue objects in PJS.
+package pjsdb
+
+import (
+	"database/sql"
+	"encoding/hex"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+const (
+	JobErrorCodeFailed       = "failed"
+	JobErrorCodeCanceled     = "canceled"
+	JobErrorCodeDisconnected = "disconnected"
+)
+
+type JobID uint64
+
+// QueueID is spec hash for now.
+type QueueID []byte
+
+// FilesetHandle is a type alias that exists purely for readability.
+type FilesetHandle []byte
+
+// Job is the internal representation of a job.
+type Job struct {
+	JobRow
+	// additional fields or metadata that would be computed would go here.
+}
+
+// JobRow models a single row in the pjs.jobs table.
+type JobRow struct {
+	ID     JobID         `db:"id"`
+	Parent sql.NullInt64 `db:"parent"`
+
+	Input    *FilesetHandle `db:"input"`
```

Comment:
Given that `FilesetHandle` is a slice, is there a reason for the double-indirection? Is a zero-length byte slice different from nil here?